qcfractal/interface/models/task_models.py

```diff
@@ -37,36 +47,83 @@ class BaseResultEnum(str, Enum):
 
 
 class PythonComputeSpec(BaseModel):
-    function: str
-    args: List[Any]
-    kwargs: Dict[str, Any]
+    function: str = Schema(
+        ...,
+        description="The module and function name of a Python-callable to call. Of the form 'module.function'"
+    )
+    args: List[Any] = Schema(
+        ...,
+        description="List of positional arguments to pass into ``function`` in order they appear."
+    )
+    kwargs: Dict[str, Any] = Schema(
+        ...,
+        description="Dictionary of keyword arguments to pass into ``function``."
+    )
 
 
 class TaskRecord(BaseModel):
 
-    id: ObjectId = None
-
-    spec: PythonComputeSpec
-    parser: str
-    status: TaskStatusEnum = "WAITING"
+    id: ObjectId = Schema(
+        None,
+        description="The Database assigned ID of the Task, if it has been assigned yet."
+    )
+
+    spec: PythonComputeSpec = Schema(
+        ...,
+        description="The Python function specification this task will execute or has executed."
+    )
+    parser: str = Schema(
+        ...,
+        description="The type of operation this is Task is. Can be 'single' or 'optimization'"
+    )
+    status: TaskStatusEnum = Schema(
+        TaskStatusEnum.waiting,
+        description="What stage of processing this task is at."
+    )
 
     # Compute blockers and prevention
-    program: str
-    procedure: Optional[str] = None
-    manager: Optional[str] = None
+    program: str = Schema(
+        ...,
+        description="Name of the quantum chemistry program which must be present to execute this task."
+    )
+    procedure: Optional[str] = Schema(
+        None,
+        description="Name of the procedure the compute platform must be able to perform to execute this task."
+    )
+    manager: Optional[str] = Schema(
+        None,
+        description="The name of the Queue Manager which has ownership of this Task"
```

Comment:
"The QueueManager that evaluated this task.". Ok to use past tense here as this field is only changed once touched by a manager.